### ISSUE TITLE
Conda package via conda-forge

### ISSUE THREAD

**Issue by almarklein** [SuperElastix/SimpleElastix#34]
It would be awesome if there were conda packages for SimpleElastix, so that anyone could just use it without needing to compile it. If done via [conda-forge](https://github.com/conda-forge), you'd only need to create (and maintain) a recipe, and the building would all be automated.

**Comment by kaspermarstal** [comment]
Hi almarklein, yes that would be very helpful for a lot of users. I do not have much experience with conda-forge, would you be interested in contributing such a recipe?

**Comment by almarklein** [comment]
I don't have a lot of time on my hands, but I do believe the Python community needs a tool like this to be easily available. What if I did the initial setup, and then ping you to sort out the details of the `build.sh` and `build.bat`?

**Comment by kaspermarstal** [comment]
That would be great! Awesome! Got your ping, will take a look this afternoon.

**Comment by almarklein** [comment]
I gave you push-rights to the repo, in case you feel like fixing things :)  I have no idea where to start with the Windows `.bat` file.

Links for reference:
* repo containing recipe (simpleelastix branch): https://github.com/almarklein/staged-recipes/tree/simpleelastix
* conda-forge PR: https://github.com/conda-forge/staged-recipes/pull/324

**Comment by almarklein** [comment]
BTW: what versions of Python does SimpleElastix' wrapper support?

**Comment by kaspermarstal** [comment]
For sure Python 2.7 and 3.4. Haven't tested with 3.5 but would be surprised if it didnt work

**Comment by kaspermarstal** [comment]
I think we might have a problem with the build time. I don't experience with Circle CI or AppVeyor, but Travis CI limits build time to 50 minutes which is barely enough time to build ITK and elastix, let alone SimpleElastix. Don't know what to do about that

**Comment by kaspermarstal** [comment]
Thanks for push-rights btw, I will handle the Windows build :)

**Comment by chvlyl** [comment]
I wonder if SimpleElastix can be installed by anaconda? 
I added conda-forge as a new channel but can't find SimpleElastix.
Here is my code:
conda config --add channels conda-forge 
conda install SimpleElastix

I got the following error message:
PackageNotFoundError: Package not found: '' Package missing in current osx-64 channels: 
  - simpleelastix

Is it because my mac OSX version (10.9.5)?

Any help is appreciated. Thanks!

**Comment by almarklein** [comment]
@chvlyl The package is not yet available. It turned out to be a lot of work to create the recipe, with problems that include that the build takes longer than the maxium allowed time for a CI job. 

@kaspermarstal I know you've already put a lot of effort in. What is the latest status, more or less? There seem to be plenty of ppl interested in this, maybe we can do a call-out for help, e.g. in the scikit-image community.

**Comment by kaspermarstal** [comment]
I met with @blowekamp at a conference yesterday who suggested to strip out the SimpleITK filters to reduce compilation time. I am working on this atm and it looks like this does indeed bring the compilation time down to under an hour. Hope to commit the changes in the next couple of days!

**Comment by blowekamp** [comment]
Hi,

It was nice meeting you. I am still at the conference, if you want to meet up to discuss this issue further.

Here is our condo-forge pull request, with some ideas on how we are going to move forward:
  https://github.com/conda-forge/staged-recipes/pull/1045

Away to help choose the components too keep to is to take advantage of ITK modularity. Has Elastix its self taken advantage of the ITK modularity? Does it build ITK with all the default modules on? A list of modules that Elastix need could be [generated](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/Maintenance/WhatModulesITK.py). And then the build of ITK could only specify the required modules.

A feature that is coming to SimpleITK, is that in each JSON file is going to specify the ITKModule it is located in. Then when building against ITK, the JSON will automatically be disabled if the ITKModule is not enabled.

Yes another way to reduce the size is to disable some pixel types, perhaps std::complex or LabelMap are not needed.

**Comment by kaspermarstal** [comment]
Hi Bradley, yes nice to meet you too! I left this morning unfortunately, would have liked to talk again. But thanks for the suggestions. Build time is dropping slowly but steady.

**Comment by kaspermarstal** [comment]
Hi @almarklein. I never got it working because it takes too long to build. Closing because it has taken too long to get it to work for me. Feel free to reopen if you want to have another go at it.

**Comment by PertuyF** [comment]
@kaspermarstal could independant [itk](https://github.com/conda-forge/itk-feedstock) and [libitk](https://github.com/conda-forge/libitk-feedstock) conda packages be of interest to speed up the build?

**Comment by cancan101** [comment]
@PertuyF that is the approach that pyfftw takes where pyftw depends on fftw (https://github.com/conda-forge/pyfftw-feedstock/blob/master/recipe/meta.yaml#L27) so that the build is broken up.

**Comment by hmaarrfk** [comment]
This is really exciting! as I'm having alot of trouble installing things on Fedora 28.

Is it possible to reopen this issue in light of recent enthusiasm?

**Comment by kaspermarstal** [comment]
Yes! I am planning to fix this build as soon as I get the SimpleITK v1.1.0 branch merged.

If anyone wants to take an initial stab at the conda build you are very welcome!
